### PR TITLE
Added bottom bar navigation for various features and code improvements

### LINKED ISSUE #423: Add bottom bar for profile and inbox screens

### PR DESCRIPTION
**Pull request by ethanboren**
This pull request introduces bottom bar navigation for different sections of the application, enhances code readability, and applies standardized formatting. Below is a summary of the main changes:

1. **Features Added:**
   - Added bottom bar navigation for:
     - Provider messages
     - Provider calendar
     - Provider profile
     - Seeker profile
     - `CreateRequest.kt` and `RequestsDashboardScreen.kt` screens

2. **Refactoring:**
   - Moved methods used in multiple places into a separate file to improve reusability and maintainability.

3. **Styling:**
   - Applied formatting using `KTFMTFormat` to ensure consistent code style.

Related Issues:
- Fixes #423 

**How to test:**
- Verify that all navigation features work correctly on the respective screens.
- Ensure formatting adheres to the project's standards.

### CODE REVIEW

File: app/src/main/java/com/android/solvit/shared/ui/authentication/SignIn.kt [hunk 1]
```diff
@@ -185,33 +176,6 @@ fun SignInScreen(
...
 }
 
-/**
- * A composable function that displays a "Go Back" button with a debounce mechanism to prevent
- * multiple rapid clicks.
- *
- * @param navigationActions A set of navigation actions to handle screen transitions.
- */
-@Composable
-fun GoBackButton(navigationActions: NavigationActions) {
```
Comment (aoutz) [inline app/src/main/java/com/android/solvit/shared/ui/authentication/SignIn.kt:195]: Great that you removed these functions from here 💪

**Review by aoutz — APPROVED** [review]
### Overall Impression:
This pull request effectively introduces bottom bar navigation across various sections of the application while focusing on code readability, reusability, and styling consistency. I've put a comment regarding a possible bug I'll let you check it out.
Overall Great job Ethan you did great! 👏

File: app/src/main/java/com/android/solvit/shared/ui/utils/ValidationRegex.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ *     - Format: "Jean-Luc", "Marie O'Connor", "José".
+ * - DESCRIPTION_REGEX: Validates general descriptions, allowing alphanumeric characters, accented
+ *   letters, spaces, and basic punctuation marks (`,`, `.`, `!`, `?`, `-`). The description can be
+ *   1 to 500 characters long.
+ *     - Format: "This is a great product!", "L'art de vivre, 100% naturel!".
+ */
+object ValidationRegex {
+  val PHONE_REGEX = Regex("^[+]?[0-9]{6,15}$")
+  val EMAIL_REGEX = Regex("^[\\w.-]+@[\\w.-]+\\.[a-zA-Z]{2,}$")
+  val FULL_NAME_REGEX = Regex("^[a-zA-Z]+(?:[-' ][a-zA-Z]+)* [a-zA-Z]+$")
```
[the remark this answers is not in the document]
Comment (ethanboren) [reply app/src/main/java/com/android/solvit/shared/ui/utils/ValidationRegex.kt:31]: Yes, I saw this typo. Thanks for your comment. I fixed it on another branch that will be merged soon.

File: app/src/main/java/com/android/solvit/seeker/ui/request/CreateRequest.kt [hunk 3]
```diff
@@ -84,140 +89,151 @@ fun CreateRequestScreen(
...
-          currentStep = 3 // Move to the analysis complete step
-        },
-        onClose = {
-          showMultiStepDialog = false
-          currentStep = 1 // Reset step for the next time
-        })
-  }
+  Scaffold(
+      bottomBar = {
+        val currentRoute = navigationActions.currentRoute() ?: "default_route"
```
Comment (manunam2002) [inline app/src/main/java/com/android/solvit/seeker/ui/request/CreateRequest.kt:94]: I don't think a bottomBar would make sense in this screen, was it explicitly decided that there should be one?

File: app/src/main/java/com/android/solvit/provider/ui/request/RequestsDashboardScreen.kt
Comment (manunam2002) [inline app/src/main/java/com/android/solvit/provider/ui/request/RequestsDashboardScreen.kt:1]: Here also I don't think we should have the bottomBar

**Review by manunam2002 — APPROVED** [review]
### Feedback

This PR adds the BottomBar where it was missing, and refracts some code to make the project more coherent and organised, great job! 
Please look at my comments: I feel like the bottom bar should be only in the screens that are on it, don't you think?

File: app/src/main/java/com/android/solvit/seeker/ui/request/CreateRequest.kt (hunk 3 above)
Comment (ethanboren) [reply app/src/main/java/com/android/solvit/seeker/ui/request/CreateRequest.kt]: Initially, I had to put a bottom bar navigation on all the screens that are on the bottom bar navigation, but you're right it's more beautiful without it, I just removed it

**Pull request merged**